scripts/run_smoke_tests.py

```
@@ -994,6 +1037,27 @@ def do_GET(self) -> None:
+        os.close(upload_fd)
+        upload_source = Path(upload_name)
+        upload_source.write_bytes(b"student-os-upload")
+        previous_base_url = os.environ.get("STUDENT_OS_MINERU_AGENT_BASE_URL")
+        try:
+            os.environ["STUDENT_OS_MINERU_AGENT_BASE_URL"] = upload_server.base_url
+            task_id = "manual-put"
+            upload_server.tasks[task_id] = {"filename": "manual.pdf", "content": b"", "headers": ""}
+            materials_convert.http_put_file(f"{upload_server.base_url.rsplit('/api/v1/agent', 1)[0]}/upload/{task_id}", upload_source)
+            upload_headers = json.loads(str(upload_server.tasks[task_id]["headers"]))
+            if "Content-Type" in upload_headers:
+                raise AssertionError(f"MinerU Agent signed upload must not send Content-Type, got: {upload_headers}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)